### ISSUE TITLE
ROME 1.6 seems to not work on Android

### ISSUE THREAD

**Issue by morphet81** [rometools/rome#294]
I recently started to use ROME and jumped directly to 1.6. Therefore, when trying to get a feed I always get the same error
`com.rometools.rome.io.ParsingFeedException: Invalid XML
[...]
Caused by: org.jdom2.JDOMException: http://apache.org/xml/features/disallow-doctype-decl feature not recognized for SAX driver org.apache.harmony.xml.ExpatReader`

I used some sample codes found on the web, but all are using older versions of ROME tools. As they actually, I tried to use the same RSS feeds they use, I've got the same error. Note that every working example that I found used FeedFetcher.

**Comment by PatrickGotthard** [comment]
ROME was never officially tested on Android. Seems like the Android XML Reader does not support the property above. The property was introduced in v1.5.1 to fix a security vulnerabilty.

**Comment by mishako** [comment]
Thank you for the report @morphet81! I looked into it and it seems to be a bug in Rome that was introduced in 2bc58a0. The change in the order of calls to `parser.setFeature` and `saxBuilder.setFeature` is what broke the code. Can't blame Patrick for that though, because original authors of that code did nothing to indicate that the order matters and prevent this from happening.

The fix is simple: swap around the lines 373 and 374 in `WireFeedInput`. Feel free to do it (preferably with a test to make sure it doesn't happen again).

So why does the order matter? Because it's the call to `parser.setFeature` that throws a `SAXNotRecognizedException` exception, i.e. tells us whether the feature is recognized or not. We don't want to call `saxBuilder.setFeature` with an unrecognized feature, because it will silently accept it and throw a `JDOMException` later in `SAXBuilder.build` method. The problem is that at that point we've already lost the context and the only way to figure out what's going on is to parse the message of the exception. We don't want to rely on exception message though.

So basically we're abusing `parser.setFeature` to find out if the feature is recognized and supported before we set it on the `saxBuilder`. Key word here is "before" :)

**Comment by mishako** [comment]
@PatrickGotthard @morphet81 Do we want to realease this bugfix as version `1.6.1`?

**Comment by mishako** [comment]
@PatrickGotthard Any thoughts on releasing `1.6.1`?

**Comment by PatrickGotthard** [comment]
Hi @mishako,

I just released version 1.6.1. It will be available in a few hours on Maven central. Would you like to update the README file and label the fixed issues with the release version?

Regards,
Patrick

**Comment by mishako** [comment]
@PatrickGotthard Looks like I don't have rights to create milestones.

**Comment by PatrickGotthard** [comment]
@mishako You now have admin rights. You proved that you do a great job and of course I have very limited time and we need another (active) admin :+1:
